src/ast/path.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <string>
+
+namespace path
```

Comment:
We're using c++17, why not use [std::filesystem](https://en.cppreference.com/w/cpp/filesystem/path)?

Reply:
filesystem support on Mac is not very good.  There are a bunch of things I implemented that can expand the platform support here.
```
get_builtin_library()
```
Has a more comprehensive `executable()` implementation, but not factored out.  And `remove_filename()` in `fs.h` covers some other bits.  

I think this can be tidied after this PR.

Reply:
Thanks for get_builtin_library() pointer, lifted some implementation from there.